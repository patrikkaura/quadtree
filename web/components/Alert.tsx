import MuiAlert from "@mui/material/Alert";

type Props = {
  foundResults: boolean;
};

export function Alert({ foundResults }: Props) {
  const message = foundResults ? "Found results" : "No results found";
  const severity = foundResults ? "success" : "warning";

  return <MuiAlert severity={severity}>{message}</MuiAlert>;
}
