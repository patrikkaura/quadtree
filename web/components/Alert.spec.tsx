import { render } from "@testing-library/react";

import { Alert } from "./Alert";

describe("Alert", () => {
  it("should match snapshot for success", () => {
    const { container } = render(<Alert foundResults={true} />);

    expect(container).toMatchSnapshot();
  });

  it("should match snapshot for failure", () => {
    const { container } = render(<Alert foundResults={false} />);

    expect(container).toMatchSnapshot();
  });
});
