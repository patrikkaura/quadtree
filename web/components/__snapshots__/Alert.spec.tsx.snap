// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert should match snapshot for failure 1`] = `
<div>
  <div
    class="flex flex-row rounded-md  gap-2 p-2 text-white bg-nord11"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="ErrorOutlineIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
      />
    </svg>
    No results found
  </div>
</div>
`;

exports[`Alert should match snapshot for success 1`] = `
<div>
  <div
    class="flex flex-row rounded-md  gap-2 p-2 text-white bg-nord14"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="CheckCircleOutlineIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M16.59 7.58 10 14.17l-3.59-3.58L5 12l5 5 8-8zM12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
      />
    </svg>
    Found results
  </div>
</div>
`;
